File: Rod Cutting Problem using Recursion/RodCuttingRecursion.py
import sys

def maxPrice(price,n):
    #base case
    if n<=0:
        return 0

    maxi = float('-inf')    
    for i in range(0,n):
        maxi = max(maxi, price[i] + maxPrice(price,n-i-1))
    
    return maxi

price = [2,5,7,8]
n = 4

print(maxPrice(price,n))

File: Rod Cutting Problem using Recursion/RodCuttingRecursion.java
import java.util.*;
public class RodCuttingRecursion
{
    static int maxPrice(int[] price,int n) {
        // base case
        if(n<=0)
            return 0;
        int maxi = Integer.MIN_VALUE;
        
        for(int i=0;i<n;i++) {
            maxi = Math.max(maxi,price[i] + maxPrice(price,n-i-1));
        }
        
        return maxi;
    }
    
	public static void main(String[] args) {
		int price[] = {2,5,7,8}; //{ 1, 5, 8, 9, 10, 17, 17, 20 };
        int n = 4;
        System.out.println(maxPrice(price,n));
	}
}
